src/components/TaxPicker.tsx

```diff
@@ -23,60 +23,47 @@ type TaxPickerProps = {
     insets?: EdgeInsets;
 
     /** Callback to fire when a tax is pressed */
-    onSubmit: () => void;
+    onSubmit: (tax: ListItem) => void;
 };
 
 function TaxPicker({selectedTaxRate = '', taxRates, insets, onSubmit}: TaxPickerProps) {
-    const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
     const {translate} = useLocalize();
     const [searchValue, setSearchValue] = useState('');
 
-    const taxRatesCount = TransactionUtils.getEnabledTaxRateCount(taxRates.taxes);
+    const taxRatesCount = TransactionUtils.getEnabledTaxRateCount(taxRates?.taxes ?? {});
     const isTaxRatesCountBelowThreshold = taxRatesCount < CONST.TAX_RATES_LIST_THRESHOLD;
 
+    const getModifiedName = (data: TaxRate, code: string) => `${data.name} (${data.value})${selectedTaxRate === code ? ` • ${translate('common.default')}` : ''}`;
+
     const shouldShowTextInput = !isTaxRatesCountBelowThreshold;
 
-    const selectedOptions = useMemo(() => {
-        if (!selectedTaxRate) {
-            return [];
-        }
+    const section = Object.entries(taxRates?.taxes ?? {}).map(([key, item]) => ({text: getModifiedName(item, key), keyForList: key, isSelected: key === selectedTaxRate}));
+    const selectedOptionKey = section.find((taxRate) => taxRate.keyForList === selectedTaxRate)?.keyForList;
 
-        return [
-            {
-                name: selectedTaxRate,
-                enabled: true,
-                accountID: null,
-            },
-        ];
-    }, [selectedTaxRate]);
+    const filteredTaxesList = useMemo(() => {
+        const trimmedSearch = searchValue.trim();
 
-    const sections = useMemo(() => {
-        const {taxRatesOptions} = OptionsListUtils.getFilteredOptions({}, {}, [], searchValue, selectedOptions, [], false, false, false, {}, [], false, {}, [], false, false, true, taxRates);
```

Comment:
@filip-solecki For filtering tax value, Why don't we use `getFilteredOptions` function like that? This also be the way we use in the tag and category page

Reply:
More information: we also use `getFilteredOptions` in other pages like tag, category to filter options.